### PR TITLE
Added families.

### PR DESCRIPTION
**Pull request by Ubaware**
Added families. ~~Not quite finished yet, but I thought I'd put it up at this stage for input.~~
Thanks BHB, Arnah, and kolackk for some family information. (Rep gain numbers and absorption of other families. Mob kill one is just a random value that makes sense though)
Only tested on local server, haven't tried with tons of members.

Things that I still haven't done ~~(will be added shortly)~~ :
* Entitlements :
  * ~~Need to modify `MapleInviteCoordinator` to take additional parameters to pass on. (Map for summon, in case it changes before the invite is answered.)~~
  * ~~I'm not sure how to do the 2x stuff since I haven't looked there in the source yet, But I will if necessary.~~ Not included.
  * ~~Check if TP/summon map is an instance, probably block if so.~~
* ~~Possibly add finer locking to `MapleFamilyEntry` if it needs it. Should be good for now though, just might be sub-optimal.~~ Separate read/write locks would only benefit `isJunior()`, which is only used during family separation.
* ~~Absorbing other families.~~
* ~~Get proper reputation cost for separation from client.~~
* ~~Maybe cleanup `MapleFamilyEntry` if it needs it. Was tired.~~ Didn't refactor, just removed a few comments/useless lines.

Also included :
* Small fix for `USE_ENFORCE_MOB_LEVEL_RANGE`. Pointed out by Lame.
* Uses proper packets for cancelling name change/world transfer.

NOT included :
* EXP/Drop family entitlements. I don't currently know the buff IDs or the masks.

### CODE REVIEW

**Comment by BHB88** [comment]
Great job!

File: src/net/server/channel/handlers/FamilyUseHandler.java [hunk 1]
```diff
@@ -21,82 +21,117 @@
...
-        if (type == 0 || type == 1) {
+        if(type == MapleFamilyEntitlement.FAMILY_REUINION || type == MapleFamilyEntitlement.SUMMON_FAMILY) {
             victim = c.getChannelServer().getPlayerStorage().getCharacterByName(slea.readMapleAsciiString());
-            if (victim != null) {
-                if (type == 0) {
-                    c.getPlayer().changeMap(victim.getMap(), victim.getMap().getPortal(0));
-                } else {
-                    victim.changeMap(c.getPlayer().getMap(), c.getPlayer().getMap().getPortal(0));
+            if(victim != null && victim != c.getPlayer()) {
+                MapleMap targetMap = victim.getMap();
```
Comment (BHB88) [inline src/net/server/channel/handlers/FamilyUseHandler.java]: Should check if players are in the same family. Send family message 67 if not

Comment (Ubaware) [reply src/net/server/channel/handlers/FamilyUseHandler.java]: Thanks, I completely overlooked that. Done.

**Comment by Ubaware** [comment]
Fixed NPC in `FamilyAddHandler` pointed out by @BHB88.

**Pull request merged**